### PR TITLE
Refactor adding and listing votes

### PR DESCRIPTION
**Pull request by karolsojko**
One thing missing, but I would add it in a separate `quick-win` is the fact that when you vote you have to refresh - the vote count and upvoted highlihts are updated in the state but the view doesn't refresh I don't know why. Other than that listing votes and voting is 👌

### CODE REVIEW

File: app/components/Skill.jsx [hunk 1]
```diff
@@ -42,12 +44,22 @@ function getSkillBySlug(skills = {}, slug) {
...
+  const { actions } = props;
+  each(resources, (resource) => {
+    actions.voteList(skillId, resource.id);
+  });
+}
+
+function loadResources(props, skills) {
+  const { actions, params } = props;
+  const skill = getSkillBySlug(skills, params.slug);
+  actions.resourceList(skill.id).then((resourceAction) => { loadVotes(props, skill.id, resourceAction.resources); });
```
Comment (bognix) [inline app/components/Skill.jsx]: I believe you can skip `{...}` around `loadVotes....`

File: app/reducers/skillsReducer.js [hunk 2]
```diff
@@ -29,8 +14,25 @@ const initialState = {
   },
 };
 
+function updateVotes(votes, resourceId, vote) {
+  let existingVoteIndex = -1;
+  each(votes[resourceId], (existingVote, index) => {
```
Comment (bognix) [inline app/reducers/skillsReducer.js]: how about using https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex ?

File: app/components/Skill.jsx (hunk 1 above)
Comment (karolsojko) [inline app/components/Skill.jsx]: 👍

File: app/reducers/skillsReducer.js (hunk 2 above)
Comment (karolsojko) [inline app/reducers/skillsReducer.js]: Great catch!

**Pull request merged**